packages/xo-web/src/xo-app/vm/tab-general.js

```diff
@@ -194,7 +194,15 @@ const GeneralTab = decorate([
                 <Icon icon='memory' size='lg' />
               </span>
             </h2>
-            <BlockLink to={`/vms/${id}/stats`}>{statsOverview && <MemorySparkLines data={statsOverview} />}</BlockLink>
+            {vm.managementAgentDetected === true && vm.managementAgentDetected !== undefined ? (
+              <BlockLink to={`/vms/${id}/stats`}>
+                {statsOverview && <MemorySparkLines data={statsOverview} />}
+              </BlockLink>
+            ) : (
+              <BlockLink to={`/vms/${id}/stats`}>
+                <Icon color='text-warning' icon='alarm' /> {_('guestToolsNecessary')}
+              </BlockLink>
```

Comment:
You usually want to put your condition on the deepest node possible:

Do this:
```js
<Parent>
  {condition ? <Child1 /> : <Child2 />}
</Parent>
```
instead of this:
```js
{condition ? (
  <Parent>
    <Child1 />
  </Parent>
) : (
  <Parent>
    <Child2 />
  </Parent>
)}
```